Ticket-105

divitcr7 opened the pull request:
scroll to top added, cards with completed tag will come at the end, ”… New Study” ribbon will display for the studies till 30 days , Changed  heading “All Studies” to “View all studies” in learner.tsx, etc.

Ticket : https://openstax.atlassian.net/browse/RENG-105

chrisbendel commented on frontend/package.json:
```diff
@@ -67,6 +67,7 @@
     "url-parse": "^1.5.10",
     "wait-on": "^6.0.1",
     "whatwg-fetch": "^3.6.2",
+    "yarn": "^1.22.22",
```
yarn shouldn't be in here, we can remove this

chrisbendel commented on frontend/package-lock.json:
delete this file as we just want yarn.lock

chrisbendel commented on frontend/src/components/footer.tsx:
```diff
@@ -16,31 +30,86 @@ import OpenStaxURL from '../images/openstax-logo.png'
-                    </Anchor>
+        <Container bg="white" w="100%" p="md">
+            <Stack gap="md">
+                <Title
```
@divitcr7 I know the specs in the figma file probably have exact pixel values, however we want everything to appear consistent on the site. We should prefer to just use
`<Title order={X}>` and not customize the line height, font size, font weight, etc... That way other `Title order={X}` will look the same throughout the site. You can remove almost all of these properties I think

chrisbendel commented on frontend/src/components/navbar/top-navbar.tsx:
```diff
@@ -1,84 +1,143 @@
+                        <Box gap="medium" align="center">
+                            <NavbarLogoLink /> 
+                        </Box>
+                        <Box
```
instead of using <Box> we can use mantine's components like <Group> or <Stack>. these have most if not all of the properties youre using here, we want to get rid of `<Box>` from `sundry` and `boxible` and this is a good chance to do it

chrisbendel commented on frontend/src/components/navbar/top-navbar.tsx:
```diff
@@ -1,84 +1,143 @@
+            align="center"
+            gap="0px"
+            css={{
+                width: '85px',
```
this is a lot of inline styles, we should be able to simplify this more...

chrisbendel commented on frontend/src/screens/researcher/ScrollToTopButton.tsx:
```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
```
@divitcr7 take a look at the [mantine affix component](https://mantine.dev/core/affix/)

they have an example there using it for a scroll to top button - we don't need to re-invent the wheel 👍

chrisbendel commented on frontend/src/screens/learner.tsx:
```diff
@@ -145,34 +175,38 @@ export const MobileStudyCards: FC<{studies: ParticipantStudy[]}> = ({ studies })
-                    </SwiperSlide>
-                ))}
+                {studies
+                    .sort((a, b) =>
```
Maybe this can be extracted out to a function with a good name? its a little hard to read

chrisbendel commented on frontend/src/screens/learner.tsx:
```diff
@@ -184,28 +218,36 @@ export const DesktopStudyCards: FC<{studies: ParticipantStudy[]}> = ({ studies }
-                    </SwiperSlide>
-                ))}
+                {studies
+                    .sort((a, b) =>
```
Maybe this can be extracted out to a function with a good name? its a little hard to read

chrisbendel commented on frontend/src/screens/learner.tsx:
```diff
@@ -55,88 +73,100 @@ const LearnerDashboard = () => {
     return (
         <TextInput
-            w={isMobile ? '100%' : '400px'}
+            width={isMobile ? '100%' : '400px'}
```
any reason for this change?

divitcr7 commented:
Fixed css properities where hardcoded values were used, fixed 1 whitespace, didn't use box, as stated used Group & Text to make the code more understandable, fixed inline styles issues, removed scroll to top button file & used windowscroll from mantine, extracted codes to a function

chrisbendel commented on frontend/src/screens/learner.tsx:
```diff
@@ -1,49 +1,88 @@
+
+const sortAndMapStudies = (
+    studies: ParticipantStudy[],
+    renderStudy: (study: ParticipantStudy) => ReactNode
```
i think we could just return the JSX here from the template rather than passing a method to render the jsx? i think that would be simpler

like calling this method `renderStudies`

```
    return studies
        .sort((a, b) =>
            a.completedAt === b.completedAt
                ? 0
                : a.completedAt
                    ? 1
                    : -1
        )
        .map((study) => {
          return (
            <SwiperSlide style={{ padding: '1rem' }} key={study.id}>
              <StudyCard study={study} />
            </SwiperSlide>
          )
        });
```

and later in the jsx? 

```
{renderStudies(studies)}
```

Coder-Srinivas replied:
I think it can be furthur simplied by implementing the sort function in the useSearchStudies hook.

chrisbendel reviewed (approved):
Looking much better - I want to get @Coder-Srinivas to get some eyes on this as well! Thanks!

chrisbendel commented on frontend/src/screens/learner.tsx:
```diff
@@ -64,236 +103,226 @@ const LearnerDashboard = () => {
 
             <StudiesContainer />
+            {isVisible && (
+                <Box
```
@divitcr7 did you not want to use the [affix component?](https://mantine.dev/core/affix/)

Coder-Srinivas commented on frontend/src/components/footer.tsx:
The [design spec](https://www.figma.com/design/MVxS1h1qovYyDqbWf3DTyF/Version-3.0-Apr24?node-id=1706-1619&t=N6PVFDo0ypgczjsl-4) is different. The funders are more towards the center.

Coder-Srinivas commented on frontend/src/screens/learner.tsx:
```diff
@@ -64,236 +103,226 @@ const LearnerDashboard = () => {
+                <Flex justify="space-between" wrap="wrap">
+                    <StudiesTitle />
 
                     <SearchBar search={search} setSearch={setSearch} />
```
The SeachBar is below the title as per the design spec

Coder-Srinivas commented on frontend/src/screens/learner.tsx:
```diff
@@ -64,236 +103,226 @@ const LearnerDashboard = () => {
 
-                <SearchResults search={search} filteredStudies={filteredStudies} />
+                {/* Divider */}
+                <Divider
```
Divider should not be on the left, but on the right as per the design spec

Coder-Srinivas commented on frontend/src/components/navbar/top-navbar.tsx:
```diff
@@ -1,84 +1,142 @@
+                                <>
+                                    <Wrapper>
+                                        <NavbarStyledLink to="/studies" className={({ isActive }) => isActive ? 'active' : ''}>
+                            Dashboard
```
Should be named as All Studies, the active style is not per design spec, it is more thick and hugging the letters, should be spaced out.

Coder-Srinivas replied:
The navbarstyled links are not displayed on mobile

Coder-Srinivas commented on frontend/src/components/navbar/top-navbar.tsx:
```diff
@@ -1,84 +1,142 @@
+    textAlign: 'left',
+});
+
+const NavbarStyledLink = styled(NavLink)({
```
Should be underlined upon hovering as per design spec

Coder-Srinivas commented on frontend/src/components/navbar/top-navbar.tsx:
```diff
@@ -1,84 +1,142 @@
+            >
+            Hi, {user.firstName}
+            </Text>
             <Icon icon={opened ? 'chevronUp' : 'chevronDown'} />
```
Icon should be bigger as per design spec and should be aligned to the center. Tabler Icons should be used

Coder-Srinivas replied:
The arrow is also not visible for Researcher and Admin

Coder-Srinivas commented on frontend/src/components/navbar/top-navbar.tsx:
```diff
@@ -87,10 +145,10 @@ const NavMenu = () => {
 
             <Menu.Dropdown>
-                {isMobile && <Menu.Item><StyledLink to="/studies">Studies</StyledLink></Menu.Item>}
+                {isMobile && <Menu.Item><Link to="/study">Dashboard</Link></Menu.Item>}
```
Not styled as per design, achievemens are also missing

divitcr7 replied:
Achievements is being done in a different ticket

divitcr7 replied:
So this one is needed to be done in the other ticket

Coder-Srinivas commented:
OpenStax logo should be of high quality as per design spec

Coder-Srinivas commented:
Validate the items specified in the [ticket](https://openstax.atlassian.net/browse/RENG-105?atlOrigin=eyJpIjoiYjFiZDAyN2UzYTQ4NDQzZDhiN2M1OTc1OTIyMTM0NTMiLCJwIjoiaiJ9) and check the one's that are completed. You can view the checkbox if you scroll down.

Coder-Srinivas commented:
Merge Conflicts were not resolved properly. The quick link section is not visible.
